### PR TITLE
Feat: Edit on GitHub section added

### LINKED ISSUE #7776: [FEATURE] Edit page button on docs
### Description

Many good documentation follows the useful practice of including an "**Edit Docs**" or "**Create an issue**" that automatically opens the corresponding page on GitHub. This feature is incredibly convenient for users as it allows them to easily locate that page, which can be sometimes overwhelming. 

It will also encourage increased contributions to the documentation.

### Screenshots

Screenshot 2023-06-24 at 11 57 27 PM

Screenshot 2023-06-24 at 11 57 59 PM

### Additional information

Not a priority at the moment.

### PR DESCRIPTION
**Pull request by akkshitgupta**
## Fixes Issue
Closes #7776  

## Changes proposed
As proposed by [Pradumnasaraf](https://github.com/Pradumnasaraf), a section on docs route has been added that contains direct links to the GitHub's UI for users who are willing to correct typo, if any.

Moreover, two more links have been attached:
1. View page on GitHub for providing the source code for that page, in particular.
2. Create an issue to receive any feedback or to get any feature request.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

How the changes look

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8178"></a>

### CODE REVIEW

File: components/EditOnGithub.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { useRouter } from "next/router";
+import { OutlineButton } from "./Button";
```
Comment (eddiejaoude) [inline components/EditOnGithub.js:2]: I did not see `OutlineButton` as an export in the `Button` component?

File: components/EditOnGithub.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const router = useRouter();
+  const { asPath } = router;
+
+  const githubUrl = `https://github.com/EddieHubCommunity/LinkFree/edit/main/pages${
+    asPath === "/docs" ? `/docs/index.js` : `${asPath}.mdx`
+  }`;
+
+  return (
+    <div className="mt-12 border border-primary-low-medium rounded-lg p-4">
+      <a href={githubUrl} target="_blank" rel="noopener noreferrer">
```
Comment (eddiejaoude) [inline components/EditOnGithub.js:14]: Can our `Link` component be used here?

File: components/EditOnGithub.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        <OutlineButton chilrdren={"Edit page on GitHub"} />
```
Comment (eddiejaoude) [inline components/EditOnGithub.js:15]: Is this a typo?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good 👍  I left some inline comments

Have you looked at Tailwind components for the 2nd side nav? (we have access to these paid components, let me know if you want the code) https://tailwindui.com/components/application-ui/navigation/vertical-navigation#component-c62ff54939dc62098de8d38ebd671ec3

File: components/EditOnGithub.js (hunk 1 above)
Comment (akkshitgupta) [reply components/EditOnGithub.js:2]: I forgot to push Button.js changes

File: components/EditOnGithub.js (hunk 2 above)
Comment (akkshitgupta) [reply components/EditOnGithub.js:14]: I tried Link component for the same but dropped the idea due to less visibility for the text in the dark mode. I was about to propose this problem.

**Comment by akkshitgupta** [comment]
Great idea!! I checked the side nav you mentioned. It suites our requirement would be great if we can implement that.

**Comment by eddiejaoude** [comment]
> Great idea!! I checked the side nav you mentioned. It suites our requirement would be great if we can implement that.

Perfect! This is the example code from their docs for that component, if you need anything else let me know

```js
import {
  CalendarIcon,
  ChartPieIcon,
  DocumentDuplicateIcon,
  FolderIcon,
  HomeIcon,
  UsersIcon,
} from '@heroicons/react/24/outline'

const navigation = [
  { name: 'Dashboard', href: '#', icon: HomeIcon, count: '5', current: true },
  { name: 'Team', href: '#', icon: UsersIcon, current: false },
  { name: 'Projects', href: '#', icon: FolderIcon, count: '12', current: false },
  { name: 'Calendar', href: '#', icon: CalendarIcon, count: '20+', current: false },
  { name: 'Documents', href: '#', icon: DocumentDuplicateIcon, current: false },
  { name: 'Reports', href: '#', icon: ChartPieIcon, current: false },
]
const secondaryNavigation = [
  { name: 'Website redesign', href: '#', initial: 'W', current: false },
  { name: 'GraphQL API', href: '#', initial: 'G', current: false },
  { name: 'Customer migration guides', href: '#', initial: 'C', current: false },
  { name: 'Profit sharing program', href: '#', initial: 'P', current: false },
]

function classNames(...classes) {
  return classes.filter(Boolean).join(' ')
}

export default function Example() {
  return (
    <nav className="flex flex-1 flex-col" aria-label="Sidebar">
      <ul role="list" className="flex flex-1 flex-col gap-y-7">
        <li>
          <ul role="list" className="-mx-2 space-y-1">
            {navigation.map((item) => (
              <li key={item.name}>
                <a
                  href={item.href}
                  className={classNames(
                    item.current
                      ? 'bg-gray-50 text-indigo-600'
                      : 'text-gray-700 hover:text-indigo-600 hover:bg-gray-50',
                    'group flex gap-x-3 rounded-md p-2 text-sm leading-6 font-semibold'
                  )}
                >
                  <item.icon
                    className={classNames(
                      item.current ? 'text-indigo-600' : 'text-gray-400 group-hover:text-indigo-600',
                      'h-6 w-6 shrink-0'
                    )}
                    aria-hidden="true"
                  />
                  {item.name}
                  {item.count ? (
                    <span
                      className="ml-auto w-9 min-w-max whitespace-nowrap rounded-full bg-white px-2.5 py-0.5 text-center text-xs font-medium leading-5 text-gray-600 ring-1 ring-inset ring-gray-200"
                      aria-hidden="true"
                    >
                      {item.count}
                    </span>
                  ) : null}
                </a>
              </li>
            ))}
          </ul>
        </li>
        <li>
          <div className="text-xs font-semibold leading-6 text-gray-400">Projects</div>
          <ul role="list" className="-mx-2 mt-2 space-y-1">
            {secondaryNavigation.map((item) => (
              <li key={item.name}>
                <a
                  href={item.href}
                  className={classNames(
                    item.current
                      ? 'bg-gray-50 text-indigo-600'
                      : 'text-gray-700 hover:text-indigo-600 hover:bg-gray-50',
                    'group flex gap-x-3 rounded-md p-2 text-sm leading-6 font-semibold'
                  )}
                >
                  <span
                    className={classNames(
                      item.current
                        ? 'text-indigo-600 border-indigo-600'
                        : 'text-gray-400 border-gray-200 group-hover:border-indigo-600 group-hover:text-indigo-600',
                      'flex h-6 w-6 shrink-0 items-center justify-center rounded-lg border text-[0.625rem] font-medium bg-white'
                    )}
                  >
                    {item.initial}
                  </span>
                  <span className="truncate">{item.name}</span>
                </a>
              </li>
            ))}
          </ul>
        </li>
      </ul>
    </nav>
  )
}
```

**Comment by akkshitgupta** [comment]
result

video of final result

**Comment by akkshitgupta** [comment]
@eddiejaoude sir can you please look at the changes made.

File: components/EditOnGithub.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import { useRouter } from "next/router";
+import { RiCodeSSlashFill, RiEdit2Fill, RiFeedbackLine } from "react-icons/ri";
+
+const className =
+  "hover:bg-primary-low group hover:text-secondary-high dark:hover:text-secondary-high flex w-full text-left rounded-md p-2 gap-x-3 text-sm leading-6 font-semibold text-primary-high dark:text-primary-low-medium";
+
+const iconClass = "h-5 w-5 self-center";
```
Comment (eddiejaoude) [inline components/EditOnGithub.js:7]: I think these constants should go in the function below also

**Review by eddiejaoude — APPROVED** [review]
Great work 👍  thank you!

I made an inline comment but this change can come in a future PR, as I am keen to get this new feature (and value) out to the users

**Comment by akkshitgupta** [comment]
Glad to hear that. I would like to make asap with other changes included in that

**Comment by eddiejaoude** [comment]
> Glad to hear that. I would like to make asap with other changes included in that

Great 👍  thank you

File: components/EditOnGithub.js (hunk 4 above)
Comment (akkshitgupta) [reply components/EditOnGithub.js:7]: Should these changes be made in a new PR or is it okay to make the changes in the PR which fix the issue #8046

Comment (eddiejaoude) [reply components/EditOnGithub.js:7]: That is a separate area of the code, I would make a separate PR for these changes

**Pull request merged**